08-regression2.Rmd

```diff
@@ -453,41 +463,94 @@ get our test error.
-backward or forward selection, and L1 or L2 regularization (also known as Lasso
-and ridge regression, respectively).
+As mentioned earlier, this is not always the case: sometimes including more
+predictors can negatively impact the prediction performance on unseen 
+test data (here we compare test errors purely for the purposes of teaching).
+
+## Designing new predictors
+
+We were quite fortunate in our initial exploration to find a predictor variable (house size)
+that seems to have a meaningful and nearly linear relationship with our response variable (sale price).
+But what should we do if we cannot immediately find such a nice variable?
+Well, sometimes it is just a fact that the variables in the data do not have enough of 
+a relationship with the response variable to provide useful predictions. For example,
+if the only available predictor was "the current house owner's favourite icecream flavour",
+we likely would have little hope of using that variable to predict the house's sale price
+(barring any future remarkable scientific discoveries about the relationship between 
+the housing market and homeowner icecream preferences). In cases like these, 
+the only option is to obtain measurements of more useful variables.
+
+There are, however, a wide variety of cases where the predictor variables do have a 
+meaningful relationship with the response variable, but that relationship does not fit
+the assumptions of the regression method you have chosen. For example, a dataframe `df`
+with two variables---`x` and `y`---with a nonlinear relationship between the two variables
+will not be correctly captured by simple linear regression:
+<center>
+```{r 08-predictor-design, message = FALSE, warning = FALSE, echo = FALSE, fig.height = 4, fig.width = 5, fig.cap = "Example of a dataset with a nonlinear relationship between the predictor and the response"}
+set.seed(1)
+df = tibble(x = sample(10000, 100, replace=TRUE)/10000)
+df <- df %>% mutate(y = x^3 + 0.2*sample(10000, 100, replace=TRUE)/10000 - 0.1)
+curve_plt <- ggplot(df, aes(x = x, y = y)) +
+  geom_point() +
+  xlab("x") +
+  ylab("y") +
+  geom_smooth(method = "lm", se = FALSE)
+curve_plt
+```
+</center>
+Instead of trying to predict the response `y` using a linear regression on `x`,
+we might have some scientific background about our problem to suggest that `y`
+should be a cubic function of `x`. So before performing regression,
+we might *create a new predictor variable* `z` using the `mutate` function:
+```{r 08-predictor-design-3, message = FALSE, warning = FALSE}
+df <- df %>%
+        mutate(z = x^3)
+```
+Then we can perform linear regression for `y` using the predictor variable `z`:
+<center>
+```{r 08-predictor-design-2, message = FALSE, warning = FALSE, echo = FALSE, fig.height = 4, fig.width = 5, fig.cap = "Relationship between the transformed predictor and the response"}
+curve_plt2 <- ggplot(df, aes(x = z, y = y)) +
+  geom_point() +
+  xlab("z") +
```

Comment:
I think we should make it clear here that z = x^3 (easy for a learner to forget this key thing when they are learning).
```suggestion
  xlab("z (x^3)") +
```

Reply:
done